### PR TITLE
feat(react-infobutton): Add InfoIcon's implementation

### LINKED ISSUE #28047: InfoIcon: Add image + tooltip variant
After an offline talk, we've decided to go with adding both an infobutton and an InfoIcon, this is tracking the addition of InfoIcon

### PR DESCRIPTION
**Pull request by sopranopillow**
This PR adds the implementation of InfoIcon.

InfoIcon is another use case of an info bubble, we currently have an InfoButton that requires the user to click a button to get the information and InfoIcon provides focus/hover toggle behavior. InfoButton is useful when there's focusable items while InfoIcon is useful for short text and when there's no focusable elements.

## Related Issue(s)

- Fixes #28047

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/Avatar-mount_base.html">617</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/Avatar-mount.html">585</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/Button-mount_base.html">302</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/Button-mount.html">303</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/Field-mount_base.html">1042</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/Field-mount.html">1042</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/FluentProvider-mount_base.html">647</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/FluentProvider-mount.html">681</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/FluentProviderWithTheme-mount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/FluentProviderWithTheme-mount.html">88</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">64</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">66</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">76</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/InfoButton-mount_base.html">15</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/InfoButton-mount.html">11</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/MakeStyles-mount_base.html">873</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/MakeStyles-mount.html">862</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/Persona-mount_base.html">1609</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/Persona-mount.html">1583</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/SpinButton-mount_base.html">1296</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28105/perf-test-react-components/SpinButton-mount.html">1287</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-infobutton/src/components/InfoButton/useInfoButtonStyles.styles.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ const useButtonStyles = makeStyles({
 
     backgroundColor: tokens.colorTransparentBackground,
     color: tokens.colorNeutralForeground2,
+    cursor: 'pointer',
```
Comment (sopranopillow) [inline packages/react-components/react-infobutton/src/components/InfoButton/useInfoButtonStyles.styles.ts:30]: To differentiate between the InfoButton and InfoTip, we're adding the cursor pointer to make it easier to tell InfoButton has a button.

File: packages/react-components/react-infobutton/src/components/InfoTip/useInfoTip.ts [hunk 2]
```diff
@@ -12,17 +14,40 @@ import type { InfoTipProps, InfoTipState } from './InfoTip.types';
...
+      root: 'span',
+      tooltip: Tooltip as React.FC<Partial<TooltipProps>>,
     },
-    // TODO add appropriate slots, for example:
-    // mySlot: resolveShorthand(props.mySlot),
-    root: getNativeElementProps('div', {
-      ref,
+
+    root: getNativeElementProps('span', {
+      'aria-label': 'information',
```
Comment (smhigley) [inline packages/react-components/react-infobutton/src/components/InfoTip/useInfoTip.ts:44]: I think I'd lean towards not including this by default, and having docs suggesting adding a short one-word `aria-label` with something similar to "information".

The reason being that I think the best word to use is a bit contextual (even aside from localization), and including an extra word before the tooltip text is helpful context but not actually necessary

**Review by smhigley — COMMENTED** [review]
Couple minor things, overall it's looking good!

File: packages/react-components/react-infobutton/src/components/InfoIcon/useInfoIconStyles.styles.ts [hunk 3]
```diff
@@ -1,33 +1,95 @@
...
+      color: tokens.colorNeutralForeground2BrandHover,
+
+      [`& .${iconFilledClassName}`]: {
+        display: 'inline-flex',
+      },
+      [`& .${iconRegularClassName}`]: {
+        display: 'none',
+      },
+    },
   },
```
Comment (khmakoto) [inline packages/react-components/react-infobutton/src/components/InfoIcon/useInfoIconStyles.styles.ts:42]: You should create these base styles using `makeResetStyles` for better performance.

**Pull request closed without merging**